File: Implementacion/inteligencia/conMemo/objeto.py

# **************************************************************************
#			  objeto.py  -  description
#   	                      ------------------
# Wizard's Sword
# desarrollado por     : (C) 2004 by M Beatriz Medina Yanez
# *************************************************************************


#  Clase que implementa los atributos de un objeto del juego

# id:       Identificador del objeto

# tipo:     Tipo de objeto
#           0 -> arbol
#           1 -> setas
#           2 -> miel
#           3 -> cofre		 
#           4 -> bruja
#           5 -> candelabro
#           6 -> huesos
#           7 -> portal magico
#           8 -> estatua
#           9 -> esclava1
#           10 -> esclava1 libre
#           11 -> esclava2
#           12 -> esclava2 libre
#           13 -> esclavo1
#           14 -> esclavo1 libre
#           15 -> esclavo2
#           16 -> esclavo2 libre
#           17 -> pinchos
#           18 -> bibliotecaria
#           19 -> mesa
#           20 -> biblioteca
#           21 -> posadera
#           22 -> cama
#           23 -> vela
#           24 -> escoba
#           25 -> armero
#           26 -> estatua con armadura
#           27 -> espada1
#           28 -> espada2
#           29 -> hacha
#           30 -> bola y cadena
#           31 -> escudo1
#           32 -> escudo2
#           33 -> escudo3
#           34 -> maniqui
#           35 -> armadura de cuero
#           36 -> cota de anillos
#           37 -> armadura
#           38 -> coraza

# visi:     Flag que indica si el objeto es visible o no
#           0 -> No visible
#           1 -> Visible
 
# posx:     Posicion x del objeto
# posy:     Posicion y del objeto

# longx:    Longitud x del objeto
# longy:    Longitud y del objeto

# casilx:   Numero de casillas que ocupa en el eje x
# casily:   Numero de casillas que ocupa en el eje y

# edif:     Identificador del edificio donde se encuentra el objeto
#           o VACIO si no se encuentra en ninguno

# equ:      Tipo de objeto que contiene el objeto o VACIO si no contiene nada
#           0     -> llave
#           1-10  -> pocion (i) (i: 1->10)
#           11    -> monedas (50 unidades)

# nAtr:     Numero de atributos del objeto Objeto

class Objeto:
    __id = None
    __tipo = None
    __visi = None
    __posx = None
    __posy = None
    __longx = None
    __longy = None
    __casilx = None
    __casily = None
    __edif = None	
    __equ = None
    __nAtr = None

    # Inicializador
    def __init__ (self):
        self.__nAtr = 11

    # Metodo consultor que devuelve el numero de atributos del objeto Objeto
    def obtieneNAtributos (self):
        return self.__nAtr
    
    # Metodo consultor que devuelve el valor del atributo especificado
    def obtieneAtributo (self, atributo):

        if ((atributo == 1) or (atributo == 'id')):
            return self.__id

        elif ((atributo == 2) or (atributo == 'tipo')):
            return self.__tipo

        elif ((atributo == 3) or (atributo == 'visi')):
            return self.__visi

        elif ((atributo == 4) or (atributo == 'posx')):
            return self.__posx

        elif ((atributo == 5) or (atributo == 'posy')):
            return self.__posy

        elif ((atributo == 6) or (atributo == 'longx')):
            return self.__longx

        elif ((atributo == 7) or (atributo == 'longy')):
            return self.__longy

        elif ((atributo == 8) or (atributo == 'casilx')):
            return self.__casilx

        elif ((atributo == 9) or (atributo == 'casily')):
            return self.__casily

        elif ((atributo == 10) or (atributo == 'edif')):
            return self.__edif

        elif ((atributo == 11) or (atributo == 'equ')):
            return self.__equ

    # Metodo que actualiza con un nuevo valor al atributo especificado    
    def reAsignaAtributo (self, atributo, nuevoValor):

        if ((atributo == 1) or (atributo == 'id')):
            self.__id = nuevoValor

        elif ((atributo == 2) or (atributo == 'tipo')):
            self.__tipo = nuevoValor

        elif ((atributo == 3) or (atributo == 'visi')):
            self.__visi = nuevoValor

        elif ((atributo == 4) or (atributo == 'posx')):
            self.__posx = nuevoValor

        elif ((atributo == 5) or (atributo == 'posy')):
            self.__posy = nuevoValor

        elif ((atributo == 6) or (atributo == 'longx')):
            self.__longx = nuevoValor

        elif ((atributo == 7) or (atributo == 'longy')):
            self.__longy = nuevoValor

        elif ((atributo == 8) or (atributo == 'casilx')):
            self.__casilx = nuevoValor

        elif ((atributo == 9) or (atributo == 'casily')):
            self.__casily = nuevoValor

        elif ((atributo == 10) or (atributo == 'edif')):
            self.__edif = nuevoValor

        elif ((atributo == 11) or (atributo == 'equ')):
            self.__equ = nuevoValor

# *************************************************************************

File: Implementacion/objeto.cc

/***************************************************************************
                          objeto.cc  -  description
                             -------------------
  Wizard's Sword
  desarrollado por     : (C) 2004 by Mª Beatriz Medina Yañez
 ***************************************************************************/

#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#include "objeto.hh"

/***************************************************************************/

// Constructor del objeto Objeto que inicializa todos sus atributos a VACIO
Objeto::Objeto ()
{
    // Inicialización de los atributos a VACIO
    this->id = VACIO;
    this->tipo = VACIO;
    this->visi = VACIO;
    this->posx = VACIO;
    this->posy = VACIO;
    this->longx = VACIO;
    this->longy = VACIO;
    this->casilx = VACIO;
    this->casily = VACIO;
    this->edif = VACIO;	
    this->equ = VACIO;

    // Inicialización del número de atributos
    this->nAtr = 11;
}

/***************************************************************************/

// Destructor del objeto Objeto
Objeto::~Objeto ()
{}

/***************************************************************************/

// Método consultor que devuelve el número de atributos del objeto Objeto
int Objeto::obtieneNAtributos (void)
{
    return this->nAtr;
}

/***************************************************************************/

/* Método consultor que devuelve el valor del atributo cuyo nombre se 
   especifica */ 
int Objeto::obtieneAtributo (char *atributo)
{
    if (!strcmp (atributo, "id"))
	return this->id;

    else if (!strcmp (atributo, "tipo"))
	return this->tipo;

    else if (!strcmp (atributo, "visi"))
	return this->visi;

    else if (!strcmp (atributo, "posx"))
	return this->posx;

    else if (!strcmp (atributo, "posy"))
	return this->posy;

    else if (!strcmp (atributo, "longx"))
	return this->longx;

    else if (!strcmp (atributo, "longy"))
	return this->longy;

    else if (!strcmp (atributo, "casilx"))
	return this->casilx;

    else if (!strcmp (atributo, "casily"))
	return this->casily;

    else if (!strcmp(atributo, "edif"))
	return this->edif;

    else if (!strcmp (atributo, "equ"))
	return this->equ;

    else
    {
	fprintf (stderr, "objeto.cc::obtieneAtributo (char *): ");
	fprintf (stderr, "El atributo %s no existe\n", atributo);
	exit (-1);
    }
}

/***************************************************************************/

/* Método consultor que devuelve el valor del atributo cuyo índice se 
   especifica */ 
int Objeto::obtieneAtributo (int atributo)
{
    if (atributo == 1)
	return this->id;

    else if (atributo == 2)
	return this->tipo;

    else if (atributo == 3)
	return this->visi;

    else if (atributo == 4)
	return this->posx;

    else if (atributo == 5)
	return this->posy;

    else if (atributo == 6)
	return this->longx;

    else if (atributo == 7)
	return this->longy;

    else if (atributo == 8)
	return this->casilx;

    else if (atributo == 9)
	return this->casily;

    else if (atributo == 10)
	return this->edif;

    else if (atributo == 11)
	return this->equ;

    else
    {
	fprintf (stderr, "objeto.cc::obtieneAtributo (char *): ");
	fprintf (stderr, "El atributo %s no existe\n", atributo);
	exit (-1);
    }
}

/***************************************************************************/

/* Método que actualiza con un nuevo valor al atributo cuyo nombre se
   especifica */
void Objeto::reAsignaAtributo (char *atributo, int nuevoValor)
{
    if (!strcmp (atributo, "id"))
	this->id = nuevoValor;

    else if (!strcmp (atributo, "tipo"))
	this->tipo = nuevoValor;

    else if (!strcmp (atributo, "visi"))
	this->visi = nuevoValor;

    else if (!strcmp (atributo, "posx"))
	this->posx = nuevoValor;

    else if (!strcmp (atributo, "posy"))
	this->posy = nuevoValor;

    else if (!strcmp (atributo, "longx"))
	this->longx = nuevoValor;

    else if (!strcmp (atributo, "longy"))
	this->longy = nuevoValor;

    else if (!strcmp (atributo, "casilx"))
	this->casilx = nuevoValor;

    else if (!strcmp (atributo, "casily"))
	this->casily = nuevoValor;

    else if (!strcmp(atributo, "edif"))
	this->edif = nuevoValor;

    else if (!strcmp (atributo, "equ"))
	this->equ = nuevoValor;
    
    else
    {
	fprintf (stderr, "objeto.cc::reAsignaAtributo (char *, int): ");
	fprintf (stderr, "El atributo %s no existe\n", atributo);
	exit (-1);
    }    
}

/***************************************************************************/

/* Método que actualiza con un nuevo valor al atributo cuyo índice se
   especifica */
void Objeto::reAsignaAtributo (int atributo, int nuevoValor)
{
    if (atributo == 1)
	this->id = nuevoValor;

    else if (atributo == 2)
	this->tipo = nuevoValor;

    else if (atributo == 3)
	this->visi = nuevoValor;

    else if (atributo == 4)
	this->posx = nuevoValor;

    else if (atributo == 5)
	this->posy = nuevoValor;

    else if (atributo == 6)
	this->longx = nuevoValor;

    else if (atributo == 7)
	this->longy = nuevoValor;

    else if (atributo == 8)
	this->casilx = nuevoValor;

    else if (atributo == 9)
	this->casily = nuevoValor;

    else if (atributo == 10)
	this->edif = nuevoValor;

    else if (atributo == 11)
	this->equ = nuevoValor;
    
    else
    {
	fprintf (stderr, "objeto.cc::reAsignaAtributo (char *, int): ");
	fprintf (stderr, "El atributo %s no existe\n", atributo);
	exit (-1);
    }    
}

/***************************************************************************/
